pkg/plugin/state.go

```diff
@@ -630,13 +632,20 @@ func (e *AutoscaleEnforcer) reserveResources(
 
 	e.state.lock.Lock()
 	defer e.state.lock.Unlock()
+	podName := util.GetNamespacedName(pod)
 
 	// If the pod already exists, nothing to do
-	if _, ok := e.state.pods[util.GetNamespacedName(pod)]; ok {
+	_, isPodInState := e.state.pods[podName]
+	if isPodInState {
 		logger.Info("Pod already exists in global state")
 		return true, &verdictSet{cpu: "", mem: ""}, nil
 	}
 
+	// if the pod doesn't exists, but has nodename attached log it
+	if !isPodInState && !opts.preexisting {
```

Comment:
IIRC `reserveResources()` is also called for pods that aren't assigned to this scheduler -- in those cases, it's expected that the pod isn't already in the state.